service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java

```diff
@@ -52,6 +53,17 @@
     private final PbaNumberService pbaNumberService;
     private final UploadAdditionalApplicationsService uploadAdditionalApplicationsService;
     private final ApplicationsFeeCalculator applicationsFeeCalculator;
+    private final ApplicantsListGenerator applicantsListGenerator;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        caseDetails.getData().put("applicantsList", applicantsListGenerator.buildApplicantsList(caseData));
```

Comment:
Although I would assume that it is working can you double check that you don't need to regenerate the dynamic list on mid event calls. There used to be an issue where the dynamic list would be sent back to us as just the selected item code instead of the actual list and as such would need regenerating otherwise it would fail validation in CCD. Going by the fact you are not doing this I am assuming it is now working and as such I think that the data type in case data can be changed from Object to DynamicList